packages/core/src/base.ts

```diff
@@ -175,7 +175,7 @@ export abstract class FrontendBase<B extends Backend, O extends Options>
 
     if (await this.getBackend().storeBreadcrumb(finalBreadcrumb, scope)) {
       scope.breadcrumbs = [...scope.breadcrumbs, finalBreadcrumb].slice(
-        -maxBreadcrumbs,
+        -Math.max(0, Math.min(maxBreadcrumbs, 100)),
```

Comment:
We could extract this `100` to a constant as well, but it's not necessary. I don't mind leaving it as is.